nmdc_api_utilities/nmdc_search.py

```diff
@@ -140,3 +141,133 @@ def get_linked_instances_and_associate_ids(
                     association[upstream_id].append(study_id)
 
         return association
+
+    def get_record_name_from_id(self, doc_id: str) -> str:
+        """
+        Used when you have an id but not the collection name.
+        Determine the schema class by which the id belongs to.
+
+        Parameters
+        ----------
+        doc_id: str
+            The id of the document.
+
+        Returns
+        -------
+        str
+            The collection name of the document.
+
+        Raises
+        ------
+        RuntimeError
+            If the API request fails.
+
+        """
+        url = f"{self.base_url}/nmdcschema/ids/{doc_id}/collection-name"
+        try:
+            response = requests.get(url)
+            response.raise_for_status()
+        except requests.exceptions.RequestException as e:
+            logger.error("API request failed", exc_info=True)
+            raise RuntimeError("Failed to get record name from NMDC API") from e
+        else:
+            logging.debug(
+                f"API request response: {response.json()}\n API Status Code: {response.status_code}"
+            )
+
+        collection_name = response.json()["collection_name"]
+        return collection_name
+
+    def get_records_by_id(
```

Comment:
The name of the function makes it seems like you can pass multiple IDs, but the functionality looks like just one?  It'd be great if we could do multiple ids in this function.